src/pages/index.js

```diff
@@ -31,37 +58,78 @@ const profileEditModal = new PopupWithForm(
 );
 profileEditModal.setEventListeners();
 
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: renderCard,
-  },
-  ".cards__list"
-);
+const avatarModal = new PopupWithForm("#avatar-modal", handleAvatarSubmit);
+avatarModal.setEventListeners();
+
+const deleteModal = new PopupDelete("#delete-modal");
+deleteModal.setEventListeners();
 
-section.renderItems();
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
+);
 
-const userInfo = new UserInfo(".profile__title", ".profile__description");
+api.getUserInfo().then((userData) => {
+  userInfo.setAvatar(userData);
+  userInfo.setUserInfo({
+    name: userData.name,
+    about: userData.about,
+  });
+});
 
 // Functions
 
-function createCard(data) {
-  const card = new Card(data, "#card-template", handleImageClick).getView();
-  return card;
+function renderLoading(isLoading, button) {
+  if (isLoading) {
+    button.textContent = "Saving...";
+  } else {
+    button.textContent = "Save";
+  }
 }
 
 // Event Handlers
 
 function handleProfileEditSubmit(inputValues) {
   console.log(inputValues);
-  userInfo.setUserInfo(inputValues);
+  renderLoading(true, editSubmitButton);
+  api
+    .updateUserInfo(inputValues)
+    .then((res) => {
+      userInfo.setUserInfo(res);
+    })
+    .finally(() => {
+      renderLoading(false, editSubmitButton);
+    });
   profileEditModal.close();
   profileEditValidator.disableButton();
 }
 
+function handleAvatarSubmit(inputValues) {
+  console.log(inputValues);
+  renderLoading(true, avatarPopupButton);
+  api
+    .updateAvatar(inputValues)
+    .then((res) => {
+      userInfo.setAvatar(res);
+      avatarModal.close();
+    })
+    .finally(() => {
+      renderLoading(false, avatarPopupButton);
```

Comment:
**Could be improved**


If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}

```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```